src/order/antichain.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2020 Bhavik Mehta, Alena Gusakov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta, Alena Gusakov
+-/
+import order.basic
+import data.finset
+import data.multiset.finset_ops
+
+/-!
+# Antichains
+Investigating the structure of finsets in a partial order.
+We define antichains.
+
+## Main definitions
+* `antichain` is a finset of elements in a partial order where
+  no element is less than another.
+
+* The `join` of two antichains is the set obtained by taking
+  their union and removing all elements that are less than
+  another element.
+
+* The `meet` of two antichains is the set obtained by taking
+  their intersection and removing all elements that are less than
+  another element.
```

Comment:
Does it make sense to use `\sup` and `\inf` for these operations?

Reply:
I wouldn't be opposed, though I wasn't sure if using it would make things weird since we have `\sup` and `\inf` in the finset file. Although I supposed making `antichain` a type would fix that